qiime/core/type/signature.py

```diff
@@ -94,7 +103,8 @@ def defaults(self):
             (name, spec.default) for name, spec in self.parameters.items()
             if spec.has_default()])
 
-    def _parse_signature(self, callable, inputs, parameters, outputs):
+    def _parse_signature(self, callable, inputs, parameters, outputs,
+                         help_text):
         # Copy so we can "exhaust" the collections and check for missing params
         inputs = copy.copy(inputs)
         parameters = copy.copy(parameters)
```

Comment:
It would probably be good to `copy.copy(help_text)` so that we can `pop` and see if there's anything left over at the end (these represent parameters that were forgotten/misspelled).